File: jphp-jxbrowser-wrapper/src-jvm/main/java/org/develnext/jphp/ext/jxbrowser/engine/JXBrowserEngine.java
package org.develnext.jphp.ext.jxbrowser.engine;

import com.teamdev.jxbrowser.chromium.Browser;
import com.teamdev.jxbrowser.chromium.BrowserContextParams;
import com.teamdev.jxbrowser.chromium.JSValue;
import com.teamdev.jxbrowser.chromium.javafx.BrowserView;
import javafx.scene.Node;


import javafx.scene.layout.StackPane;
import org.develnext.jphp.ext.javafx.classes.layout.UXStackPane;
import org.develnext.jphp.ext.jxbrowser.JXBrowserExtension;
import php.runtime.Memory;
import php.runtime.annotation.Reflection;
import php.runtime.env.Environment;
import php.runtime.lang.BaseWrapper;
import php.runtime.memory.DoubleMemory;
import php.runtime.memory.StringMemory;
import php.runtime.reflection.ClassEntity;

import java.util.UUID;

@Reflection.Namespace(JXBrowserExtension.NS + "\\engine")
@Reflection.Name("JXBrowserEngine")
public class JXBrowserEngine extends BaseWrapper<Browser>{

    public JXBrowserEngine(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }
    public JXBrowserEngine(Environment env, Browser wrappedObject) {
        super(env, wrappedObject);
    }

    public Browser getWrappedObject(){
        return super.getWrappedObject();
    }

    @Reflection.Getter
    public String getUrl()
    {
        return __wrappedObject.getURL();
    }

    @Reflection.Setter
    public void setUrl(String urlSet)
    {
        __wrappedObject.loadURL(urlSet);
    }

    @Reflection.Setter
    public void setZoom(double zoomLvl)
    {
        __wrappedObject.setZoomLevel(zoomLvl);
    }

    @Reflection.Getter
    public double getZoom()
    {
        return  __wrappedObject.getZoomLevel();
    }



    @Reflection.Signature
    public void loadHtml(String html, int i)
    {
        __wrappedObject.loadHTML(i, html);
    }

    @Reflection.Signature
    public void loadHtml(String html)
    {
        __wrappedObject.loadHTML(html);
    }

    @Reflection.Signature
    public void executeJavaScript(String js)
    {
        __wrappedObject.executeJavaScript(js);
    }

    @Reflection.Signature
    public Memory executeJavaScriptAndReturnValue(String js)
    {
        JSValue value = __wrappedObject.executeJavaScriptAndReturnValue(js);
        if (value.isString()){return new StringMemory(value.getStringValue());}
        else if (value.isNumber()){return new DoubleMemory(value.getNumberValue());}
        else if (value.isBoolean()){return value.getBooleanValue() ? Memory.TRUE : Memory.FALSE;}
        else if (value.isNull()){return Memory.NULL;}
        else {throw new IllegalStateException("unsupported or unwrapped return value");}
    }

    @Reflection.Signature
    public void executeJavaScript(long i, String js)
    {
        __wrappedObject.executeJavaScript(i, js);
    }
    @Reflection.Signature
    public void reload() {
        __wrappedObject.reload();
    }

    @Reflection.Signature
    public void reload(boolean ignoreCache) {
        __wrappedObject.reloadIgnoringCache();
    }

    @Reflection.Signature
    public void stop() {
        __wrappedObject.stop();
    }

    @Reflection.Signature
    public void dispose() {
        __wrappedObject.dispose();
    }

    @Reflection.Signature
    public boolean canGoBack() {
        return __wrappedObject.canGoBack();
    }

    @Reflection.Signature
    public boolean canGoForward() {
        return __wrappedObject.canGoForward();
    }

    @Reflection.Signature
    public void goBack() {
        __wrappedObject.goBack();
    }

    @Reflection.Signature
    public void goForward() {
        __wrappedObject.goForward();
    }

    @Reflection.Getter
    public String getTitle() {
        return __wrappedObject.getTitle();
    }

    @Reflection.Getter
    public String getHtml() {
        return __wrappedObject.getHTML();
    }

    @Reflection.Signature
    public void print() {
        __wrappedObject.print();
    }

    @Reflection.Setter
    public void setUserAgent(String userAgent) {
        __wrappedObject.setUserAgent(userAgent);
    }

    @Reflection.Getter
    public String getUserAgent() {
        return __wrappedObject.getUserAgent();
    }

    @Reflection.Signature
    public boolean isLoading() {
        return __wrappedObject.isLoading();
    }

    @Reflection.Signature
    public void setAudioMuted(boolean muted) {
        __wrappedObject.setAudioMuted(muted);
    }

    @Reflection.Signature
    public boolean isAudioMuted() {
        return __wrappedObject.isAudioMuted();
    }



}

File: jphp-jxbrowser-wrapper/sdk/php/gui/jxbrowser/engine/JXBrowserEngine.php
<?php
namespace php\gui\jxbrowser\engine;

use php\gui\layout\UXStackPane;
use php\gui\UXApplication;
use php\gui\UXNode;

class JXBrowserEngine
{

    /**
    * @var string
    */
    public $url;

    /**
    * @var double
    */
    public $zoom;

    /**
    * @var string
    */
    public $title;

    /**
    * @var string
    */
    public $html;

    /**
    * @var string
    */
    public $userAgent;


    /**
     * Загружает HTML-контент
     * @param int $frame id браузера(frame)
     * @param string $html HTML-код
     */
    public function loadHtml($frame, $html)
    {
    }

    /**
     * Выполняет js-код
     * @param int $frame id браузера(frame)
     * @param string $js js-код
     */
    public function executeJavaScript($frame, $js)
    {
    }

    /**
     * Выполняет js-код
     * @param string $js js-код
     */
    public function executeJavaScript($js)
    {
    }

    /**
     * Перезагружает текущую страницу в браузере
     * @param bool $ignoreCache Если true, игнорирует кеш браузера (загружает с сервера)
     */
    public function reload(bool $ignoreCache = false)
    {
    }

    /**
     * Немедленно останавливает все процессы браузера:
     * - Прерывает загрузку текущей страницы
     * - Останавливает выполнение JavaScript
     * - Отменяет текущие сетевые запросы
     *
     * ⚠️ Работает как кнопка "Стоп" (⏹) в обычном браузере.
     * После вызова браузер остаётся работоспособным.
     *
     * @return void
     *
     * @example
     * // Пример использования:
     * $engine->stop(); // Принудительная остановка загрузки
     *
     * @see $engine->dispose() Для полного уничтожения браузера
     */
    public function stop()
    {

    }

    /**
     * Полностью уничтожает браузерный инстанс и освобождает все ресурсы.
     *
     * После вызова этот объект браузера нельзя использовать!
     *
     * 🔥 Жёсткий пример:
     * $engine->dispose(); // Теперь $engine — труп
     * $engine->url = 'xsite.ru'; // Вызовет ошибку
     *
     * @return void
     *
     * @example
     * // Корректное использование:
     * $engine = new JxBrowserEngine();
     * $engine->url = 'https://example.com';
     * // ... работаем ...
     * $engine->dispose(); // Чистим за собой
     *
     * @see $engine->stop() Если нужно просто остановить загрузку
     */
    public function dispose()
    {

    }


    /**
     * Запускает системный диалог печати текущей страницы.
     * По сути, это программный аналог нажатия Ctrl+P в браузере.
     *
     * @return void
     *
     * @warning
     * - Работает только если браузерная сессия активна
     */
    public function print()
    {

    }


    /**
     * Грузится ли?
     * @return bool
     */
    public function isLoading()
    {

    }

    /**
     * @return void
     */
    public function setAudioMuted()
    {

    }

    /**
     * @return bool
     */
    public function isAudioMuted()
    {

    }






}
